File: phenomenal/ifs_50.rb
class Foo
  def test(mode)
    if mode==1 
      1
    elsif mode==2 
      2
    elsif mode==3 
      3
    elsif mode==4 
      4
    elsif mode==5 
      5
    elsif mode==6 
      6
    elsif mode==7 
      7
    elsif mode==8 
      8
    elsif mode==9 
      9
    elsif mode==10 
      10
    elsif mode==11 
      11
    elsif mode==12 
      12
    elsif mode==13 
      13
    elsif mode==14 
      14
    elsif mode==15 
      15
    elsif mode==16 
      16
    elsif mode==17 
      17
    elsif mode==18 
      18
    elsif mode==19 
      19
    elsif mode==20 
      20
    elsif mode==21 
      21
    elsif mode==22 
      22
    elsif mode==23 
      23
    elsif mode==24 
      24
    elsif mode==25 
      25
    elsif mode==26 
      26
    elsif mode==27 
      27
    elsif mode==28 
      28
    elsif mode==29 
      29
    elsif mode==30 
      30
    elsif mode==31 
      31
    elsif mode==32 
      32
    elsif mode==33 
      33
    elsif mode==34 
      34
    elsif mode==35 
      35
    elsif mode==36 
      36
    elsif mode==37 
      37
    elsif mode==38 
      38
    elsif mode==39 
      39
    elsif mode==40 
      40
    elsif mode==41 
      41
    elsif mode==42 
      42
    elsif mode==43 
      43
    elsif mode==44 
      44
    elsif mode==45 
      45
    elsif mode==46 
      46
    elsif mode==47 
      47
    elsif mode==48 
      48
    elsif mode==49 
      49
    elsif mode==50 
      50
    end 
   end 
end

def testContexts
  $i = 0
  $ITER = 1000
  $CALLS = 100
  beginning_time = Time.now
  foo = Foo.new
  until $i > $ITER do
    num = 1 + rand(50)
    j = 0
    for j in (1..$CALLS)
      foo.test(num)
    end
    $i +=1;
  end
  return (Time.now - beginning_time)*1000
end

def runs(run)
  $times = 0
  File.open("runs_ifs/50/test#{run}.csv", 'w') {|f| f.write("iteration, Elapsed time in milliseconds
")}
  for $times in 1..100
    elapsed_time = testContexts
    File.open("runs_ifs/50/test#{run}.csv", 'a') do |f|
        f.puts("#{$times}, #{elapsed_time}")
      end
  end
end

for run in 1..10
  runs(run)
end

File: Subjective-C/ifs/ifs_50.m
#import <Foundation/Foundation.h>
#include <stdlib.h>
#import "ifs_50.h" 

const static int ITER = 1000;
const static int CALLS = 100; 

@implementation Foo 
-(int) testS: (int) mode {
    if(mode==1) { 
    return  1;
    }
    else if(mode==2) {
        return 2;
    }
    else if(mode==3) {
        return 3;
    }
    else if(mode==4) {
        return 4;
    }
    else if(mode==5) {
        return 5;
    }
    else if(mode==6) {
        return 6;
    }
    else if(mode==7) {
        return 7;
    }
    else if(mode==8) {
        return 8;
    }
    else if(mode==9) {
        return 9;
    }
    else if(mode==10) {
        return 10;
    }
    else if(mode==11) {
        return 11;
    }
    else if(mode==12) {
        return 12;
    }
    else if(mode==13) {
        return 13;
    }
    else if(mode==14) {
        return 14;
    }
    else if(mode==15) {
        return 15;
    }
    else if(mode==16) {
        return 16;
    }
    else if(mode==17) {
        return 17;
    }
    else if(mode==18) {
        return 18;
    }
    else if(mode==19) {
        return 19;
    }
    else if(mode==20) {
        return 20;
    }
    else if(mode==21) {
        return 21;
    }
    else if(mode==22) {
        return 22;
    }
    else if(mode==23) {
        return 23;
    }
    else if(mode==24) {
        return 24;
    }
    else if(mode==25) {
        return 25;
    }
    else if(mode==26) {
        return 26;
    }
    else if(mode==27) {
        return 27;
    }
    else if(mode==28) {
        return 28;
    }
    else if(mode==29) {
        return 29;
    }
    else if(mode==30) {
        return 30;
    }
    else if(mode==31) {
        return 31;
    }
    else if(mode==32) {
        return 32;
    }
    else if(mode==33) {
        return 33;
    }
    else if(mode==34) {
        return 34;
    }
    else if(mode==35) {
        return 35;
    }
    else if(mode==36) {
        return 36;
    }
    else if(mode==37) {
        return 37;
    }
    else if(mode==38) {
        return 38;
    }
    else if(mode==39) {
        return 39;
    }
    else if(mode==40) {
        return 40;
    }
    else if(mode==41) {
        return 41;
    }
    else if(mode==42) {
        return 42;
    }
    else if(mode==43) {
        return 43;
    }
    else if(mode==44) {
        return 44;
    }
    else if(mode==45) {
        return 45;
    }
    else if(mode==46) {
        return 46;
    }
    else if(mode==47) {
        return 47;
    }
    else if(mode==48) {
        return 48;
    }
    else if(mode==49) {
        return 49;
    }
    else if(mode==50) {
        return 50;
    }
    return 0;
}; 

-(double) testContexts {
    NSDate *start = [NSDate date];
    for(int i = 0; i < ITER; i++) {
        int num = 1 + arc4random_uniform(50);
        for(int j=1; j<= CALLS; j++) {
            [self testS: num];
        }
    }
    NSTimeInterval timeInterval = fabs([start timeIntervalSinceNow]);
    return timeInterval*1000;
} 

-(void) runs:(int) run {
    NSMutableString *data = [NSMutableString stringWithString:@"Iterations, Elapsed time in milliseconds \n "];
    for(int times = 1; times <= 100; times++) {
        double elapsed_time = [self testContexts];
        [data appendFormat:@"%d, %f \n", times, elapsed_time];
    }
    NSArray *paths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
    NSString *documentsDirectory = [paths objectAtIndex:0];
    NSString *fileName = [NSString stringWithFormat:@"%@/runs_ifs/50/test%d.csv",documentsDirectory, run];
    NSFileManager *fileManager = [NSFileManager defaultManager];
    if (![fileManager fileExistsAtPath:fileName]) {
        BOOL success = [[NSString stringWithString: data] writeToFile:fileName atomically:YES encoding:NSUTF8StringEncoding error:nil];
        if (!success) {
            NSLog(@"Error writting file");
        }
    } else {
        NSFileHandle *fileHandle = [NSFileHandle fileHandleForWritingAtPath:fileName];
        NSData *textData = [data dataUsingEncoding:NSUTF8StringEncoding];
        [fileHandle writeData:textData];
        [fileHandle closeFile];
    }
}
@end

int main(int argc, const char * argv[]) {
    Foo *foo = [[Foo alloc] init];
    for(int run=1; run <=10; run++) {
        [foo runs:run];
     }
    return 0;
}
